nxontology_ml/model.py

```diff
@@ -0,0 +1,152 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from enum import Enum
+from itertools import islice
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from catboost import CatBoostClassifier, FeaturesData, Pool
+from sklearn.metrics import classification_report
+
+from nxontology_ml.efo import get_efo_otar_slim
+from nxontology_ml.utils import ROOT_DIR
+
+
+@dataclass
+class TrainingRecord:
+    efo_otar_slim_id: str
+    efo_label: str
+    rs_classification: str
+
+
+class FeatureType(Enum):
+    LABEL = 1
+    NUMERICAL = 2
+    CATEGORICAL = 3
+    IGNORED = 4
+
+    @classmethod
+    def from_feature_name(cls, name: str) -> "FeatureType":
+        if name == "__label__":
+            return cls.LABEL
+        if cls._is_numerical(name):
+            return cls.NUMERICAL
+        if cls._is_categorical(name):
+            return cls.CATEGORICAL
+        assert cls._is_ignored(name), f"Unknown feature: {name}"
+        return cls.IGNORED
```

Comment:
Great system for now for assigning feature type. In future we can think if it makes sense to define some sort of dataclass or pydantic model where we encode feature type as a field metadata property.